docs/source/migration_guides.md

```diff
@@ -0,0 +1,115 @@
+Migrating custom widget libraries
+=================================
+
+These are migration guides aimed specifically at developers of third-party
+widgets.
+
+Migrating from 6.0 to 7.0
+-------------------------
+
+For example migrations, see these PRs:
+ - [ipyleaflet](https://github.com/ellisonbg/ipyleaflet/pull/66)
+ - [jupyter-gmaps](https://github.com/pbugnion/gmaps/pull/166)
+
+To avoid tying your development cycle to ipywidgets, we recommend starting
+the migration on a branch and keeping that branch open until ipywidgets 7.0
+is released. 
+
+We also recommend testing the migration in a completely new notebook, rather
+than one that contains widgets that you instantiated with ipywidgets 6.0.
+
+### Updating setup.py
+
+Start by updating the dependency in your `setup.py` to the latest beta. To
+find the correct version number, go to the [releases
+page](https://github.com/jupyter-widgets/ipywidgets/releases) on Github and
+cycle through the tags until you see a tag called `7.0.0bx` or `7.0.0rcx`.
+
+### Updating package.json
+
+Next, we should update the JavaScript dependencies. The most important change
+for widget developers is that the JavaScript package for jupyter-widgets has
+been split between `@jupyter-widgets/base` and `@jupyter-widgets/control`: 
+ - `@jupyter-widgets/base` contains the base widget classes and the layout
+classes 
+ - `@jupyter-widgets/controls` contains the widget classes for the
+user-facing widgets.
+
+In your `package.json`, remove `jupyter-js-widgets` from your dependencies
+and add `@jupyter-widgets/base`. To find the correct version, go to the
+[releases page](https://github.com/jupyter-widgets/ipywidgets/releases) and
+cycle through the tags until you see a tag called
+`@jupyter-widgets/base@<version>`. Do the same for
+`@jupyter-widgets/controls` if you think you have a dependency on it (e.g. if
+you create widgets that inherit from `VBox` or `HBox` or another user-facing widget).
+
+### Updating Webpack configuration
```

Comment:
One other thing is that the requirejs configuration in the classic notebook extension should no longer have a `jupyter-js-widgets` entry - see the change at https://github.com/bloomberg/bqplot/commit/7afcb19dd8c00582db47a436576f7815467e5b50#diff-c0ff43d45628755eae045a0eec2fc1bdL21